### PR TITLE
Backport: Add support for boot environment data to be stored in the label

### PR DESCRIPTION
**Pull request by pcd1193182**
http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/appliance-build-orchestrator-pre-push/3932/console (error appears unrelated)

Manual testing (cloning the ab-pre-push VM and verifying that bootcount changed appropriately) was successful.

Note that this backport also includes the follow-up commit that fixed a small error.

Original PR: https://github.com/openzfs/zfs/pull/10009

The backport applied cleanly, except for conflicts with adjacent changes due to the presence of the OS compatability layer on master but not in 6.0

### CODE REVIEW

**Comment by pzakha** [comment]
Could you link the original review please? Also, did the backport apply cleanly?

**Comment by pcd1193182** [comment]
@pzakha Updated the original message to answer those questions.

File: tests/zfs-tests/cmd/libzfs_input_check/libzfs_input_check.c [hunk 1]
```diff
@@ -989,6 +1010,8 @@ validate_ioc_values(void)
 	CHECK(LINUX_IOC_BASE + 1 == ZFS_IOC_EVENTS_NEXT);
 	CHECK(LINUX_IOC_BASE + 2 == ZFS_IOC_EVENTS_CLEAR);
 	CHECK(LINUX_IOC_BASE + 3 == ZFS_IOC_EVENTS_SEEK);
+	CHECK(LINUX_IOC_BASE + 4 == ZFS_IOC_SET_BOOTENV);
+	CHECK(LINUX_IOC_BASE + 5 == ZFS_IOC_GET_BOOTENV);
```
Comment (pzakha) [inline tests/zfs-tests/cmd/libzfs_input_check/libzfs_input_check.c:1014]: See my previous comment regarding ioctl numbers

File: include/sys/fs/zfs.h [hunk 2]
```diff
@@ -1280,18 +1280,19 @@ typedef enum zfs_ioc {
...
 	/*
-	 * Linux - 3/64 numbers reserved.
+	 * Per-platform (Optional) - 6/128 numbers reserved.
 	 */
 	ZFS_IOC_LINUX = ('Z' << 8) + 0x80,
 	ZFS_IOC_EVENTS_NEXT,			/* 0x5a81 */
 	ZFS_IOC_EVENTS_CLEAR,			/* 0x5a82 */
 	ZFS_IOC_EVENTS_SEEK,			/* 0x5a83 */
+	ZFS_IOC_SET_BOOTENV,			/* 0x5a84 */
+	ZFS_IOC_GET_BOOTENV,			/* 0x5a85 */
```
Comment (pzakha) [inline include/sys/fs/zfs.h:1290]: I think we would want to use the same ioctl numbers as on master, to ensure compatibility between different versions of userland & kernel.
See https://github.com/openzfs/zfs/pull/10009/files#diff-f29552ad5ed97006cc80fc2b453cbe39R1302

**Pull request merged**